scripts/test.py

```diff
@@ -0,0 +1,212 @@
+
+unit_test_profiles = [
+    'test_posig_snarkless',
+    'test_postake_snarkless'
+]
+
+simple_tests = [
+    'full-test',
+    'transaction-snark-profiler -check-only',
+]
+
+integration_tests = [
+    'coda-peers-test',
+    'coda-transitive-peers-test',
+    'coda-block-production-test',
+    'coda-shared-prefix-test -who-proposes 0',
+    'coda-shared-prefix-test -who-proposes 1',
+    'coda-shared-state-test',
+    'coda-restart-node-test',
+]
+
+all_tests = simple_tests + integration_tests
+
+test_permutations = {
+    'fake_hash': ['full-test'],
+    'test_posig_snarkless': all_tests,
+    'test_postake_snarkless': simple_tests,
+    'test_postake_split_snarkless': integration_tests,
+    'test_posig': simple_tests,
+    'test_postake': simple_tests,
+    'test_postake_catchup': ['coda-restart-node-test']
+}
+
+ci_blacklist = [
+    'test_postake_split_snarkless:coda-restart-node-test'
+]
+
+def fail(msg):
+    print('ERROR: %s' % msg)
+    exit(1)
+
+def test_pattern(pattern, string):
+    return (pattern and (pattern == '*' or pattern == string))
+
+def parse_filter(pattern_src):
+    if not(pattern_src):
+        return (None, None)
+    elif pattern_src == '*':
+        return ('*', '*')
+    else:
+        parts = pattern_src.split(':')
+        if len(parts) != 2:
+            fail('invalid filter syntax')
+        [profile, test] = parts
+        return (profile, test)
+
+def filter_test_permutations(whitelist_filters, blacklist_filters):
+    whitelist_patterns = [parse_filter(filter) for filter in whitelist_filters]
+    blacklist_patterns = [parse_filter(filter) for filter in blacklist_filters]
+
+    def keep(profile, test):
+        whitelisted = all(
+            test_pattern(profile_pat, profile) and test_pattern(test_pat, test)
+            for (profile_pat, test_pat) in whitelist_patterns
+        )
+        blacklisted = any(
+            test_pattern(profile_pat, profile) and test_pattern(test_pat, test)
+            for (profile_pat, test_pat) in blacklist_patterns
+        )
+        return whitelisted and not(blacklisted)
+
+    # too complex to represent as a list expression in python
+    result = {}
+    for (profile,tests) in test_permutations.items():
+        for test in tests:
+            if keep(profile, test):
+                result[profile] = result[profile] if profile in result else []
+                result[profile].append(test)
+    return result
+
+def run(args):
+    # wraps a "wet" action, printing if dry run is set, executing otherwise
+    def wet(msg, f):
+        if args.dry_run:
+            print(msg)
+        else:
+            f()
+
+    def run_cmd(cmd, on_failure):
+        def do():
+            if subprocess.call(['bash', '-c', cmd], stdout=sys.stdout, stderr=sys.stderr) != 0:
+                on_failure()
+        wet('$ ' + cmd, do)
+
+    coda_exe_path = 'src/app/cli/src/coda.exe'
+    coda_build_path = './_build/default'
+    coda_exe = os.path.join(coda_build_path, coda_exe_path)
```

Comment:
this script can't be run from inside `src`. I don't know how much of that people do, but it's the default configuration of `make build`. It'd be nice if this script were robust to either.